File: lecture03/exercise03-A/template.cpp
// Exercise 03-A: Perfect Valid Square

#include <iostream>

using namespace std;

// Functions

bool is_perfect_square(long n) {
    // TODO: Use binary search to determine if n is a perfect valid square.
    return false;
}

// Main Execution

int main(int argc, char *argv[]) {
    long n;

    while (cin >> n) {
    	cout << (is_perfect_square(n) ? "Yes" : "No") << endl;
    }

    return 0;
}

File: lecture03/exercise03-A/template.py
#!/usr/bin/env python3

# Exercise 03-A: Perfect Square

import sys

# Functions

def is_perfect_square(n: int) -> bool:
    # TODO: Use binary search to determine if n is a perfect valid square
    return False

# Main Execution

def main():
    for n in map(int, sys.stdin):
        print('Yes' if is_perfect_square(n) else 'No')

if __name__ == '__main__':
    main()
